contracts/implementations/MultipleMinter.sol

```diff
@@ -279,4 +279,140 @@ contract MultipleMinter is
 
         ChargingInternal._chargeDcx(msg.sender, MINT_VEHICLE_OPERATION);
     }
+
+    /**
+     * @notice Mints and pairs a vehicle (with a Device Definition Id) and a synthetic device, and set permissions with SACD through a metatransaction
+     * The vehicle owner signs a typed structured (EIP-712) message in advance and submits to be verified
+     * @dev Caller must have the mint vehicle sd role
+     * @param data Input data with the following fields:
+     *  manufacturerNode -> Parent manufacturer node id of the vehicle
+     *  owner -> The new nodes owner
+     *  deviceDefinitionId -> The Device Definition Id
+     *  attrInfoPairsVehicle -> List of attribute-info pairs to be added of the vehicle
+     *  integrationNode -> Parent integration node id of the synthetic device
+     *  vehicleOwnerSig -> Vehicle owner signature hash
+     *  syntheticDeviceSig -> Synthetic Device's signature hash
+     *  syntheticDeviceAddr -> Address associated with the synthetic device
+     *  attrInfoPairsDevice -> List of attribute-info pairs to be added of the synthetic device
+     * @param sacdInput SACD input args
+     *  grantee -> The address to receive the permissions
+     *  permissions -> The uint256 that represents the byte array of permissions
+     *  expiration -> Expiration of the permissions
+     *  source -> The URI source associated with the permissions
+     */
+    function mintVehicleAndSdWithDeviceDefinitionSignAndSacd(
+        MintVehicleAndSdWithDdInput calldata data,
+        SacdInput calldata sacdInput
+    ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
+        VehicleStorage.Storage storage vs = VehicleStorage.getStorage();
```

Comment:
The mass code duplication here is unfortunate. You could extract it but I assume we just think this duplication won't be around for very long so screw it.